src/test/java/ArrayStackTest.java

```diff
@@ -1,11 +1,63 @@
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.EmptyStackException;
+
 import static org.junit.jupiter.api.Assertions.*;
+//import static org.assertj.core.api.Assertions.*;
 
 class ArrayStackTest {
 
     @Test
-    void isEmpty() {
-        assertNull(null);
+    @DisplayName("Stack starts empty and isEmpty true when size is 0.")
+    public void startEmptySizeZero() {
+        Stack<Integer> stack = new ArrayStack<>();
+        assertTrue(stack.isEmpty());
+        assertEquals(stack.size(), 0);
+    }
```

Comment:
Imo, specification names should read a little clearer than this-- it's English, not code. Further, in general, each test should test as little as possible. You want test failures to isolate an issue, having multiple assertions in a single test will always make it harder. For example here, given the test fails on `size() == 0` -- is it because the size is wrong? Or `isEmpty` is wrong and the size is actually nonzero? You have to dig to find out.

I'd probably break this down into two:
```java
@Test
@DisplayName("a new stack is empty")
void aNewStackIsEmpty() {
    assertTrue(new ArrayStack<Object>().isEmpty());
}

@Test
@DisplayName("an empty stack has a size of 0")
void emptyStackHasSizeZero() {
    ArrayStack<Object> classUnderTest = new ArrayStack<>();
    assumeTrue(classUnderTest.isEmpty());
    assertEquals(0, classUnderTest.size());
}
```

I use an [assumption](https://junit.org/junit5/docs/current/user-guide/#writing-tests-assumptions) to isolate errors to either the initialization (new stack is empty), or the size agreeing with empty-ness. Note this is different from _a new stack has size 0_, it's actually _**testing the interface, not the implementation**_. Though, the first test is testing implementation details (as it must at some point, since `Stack` cannot be created).

Reply:
:pog: